core/trino-main/src/main/java/io/trino/server/CoordinatorModule.java

```diff
@@ -363,7 +363,11 @@ List<OutputStatsEstimatorFactory> getCompositeOutputDataSizeEstimatorDelegateFac
         install(new QueryExecutionFactoryModule());
 
         // cleanup
-        binder.bind(ExecutorCleanup.class).asEagerSingleton();
+        executorServiceCleanupBinder(binder)
```

Comment:
are we now going to centralize this registration? That is, if we have an executor that we want to cleanly shutdown in say troubleshooting module, we'd need to move the binding here?

if I misunderstood, then calling 'executorServiceCleanupBinder' would still work when called in many places? (in terms of eager singleton and such).

Will this be added in worker module as well?

The biggest problem we see is in test code. I'm missing a code sample or a text explanation that would show how does it improve things for tests?

Does this technique help with code outside of trino OSS (also plugins)?

Reply:
> troubleshooting module

What is "troubleshooting module"? I cannot find it in Trino.

> are we now going to centralize this registration?

No.

> then calling 'executorServiceCleanupBinder' would still work when called in many places?

Yes.

> Will this be added in worker module as well?

If needed in can be added there too.

> Does this technique help with code outside of trino OSS (also plugins)?

I think so.

> The biggest problem we see is in test code. I'm missing a code sample or a text explanation that would show how does it improve things for tests?

Added test. Thanks.